### PR TITLE
tests(smoke): remove external domains from perf-preload

### PR DESCRIPTION
**Pull request by connorjclark**
This PR removes dependencies on external domains (google font domains) for the `perf-preload` smoke test. By request of @exterkamp :)

I needed to introduce a third static server to pull off a 1:1 transition, so in the process I also cleaned up that code a bit.

### CODE REVIEW

File: cli/test/fixtures/perf/preload_tester.js [hunk 1]
```diff
@@ -12,12 +12,12 @@ document.write('<script src="/perf/level-2.js?warning&delay=500"></script>');
 // delay our preconnect-candidates so that they're not assumed to be working already
 setTimeout(() => {
   // load another origin
-  fetch('http://localhost:10503/preload.html');
+  // fetch('http://localhost:10503/preload.html');
```
Comment (connorjclark) [inline cli/test/fixtures/perf/preload_tester.js:15]: Oh, I forgot about this. I think I can just drop this line?

**Comment by adamraine** [comment]
Shot in the dark, but does this work on devtools? We're skipping it right now in devtools CI.

**Comment by connorjclark** [comment]
> Shot in the dark, but does this work on devtools? We're skipping it right now in devtools CI.

nope
image

File: core/test/fraggle-rock/scenarios/pptr-test-utils.js [hunk 2]
```diff
@@ -38,8 +38,8 @@ function createTestState() {
 
     installSetupAndTeardownHooks() {
       before(async () => {
-        this.server = new Server();
-        this.secondaryServer = new Server();
+        this.server = new Server(0);
+        this.secondaryServer = new Server(0);
```
Comment (adamraine) [inline core/test/fraggle-rock/scenarios/pptr-test-utils.js:42]: Might be nice to use `createServers` here and expose a `this.servers` for pptr tests.

File: cli/test/fixtures/preload.html [hunk 3]
```diff
@@ -4,9 +4,9 @@
   <!-- WARN(preload): This should be no crossorigin -->
   <link rel="preload" href="/perf/level-2.js?warning&delay=500" as="script" crossorigin="use-credentials"/>
   <!-- WARN(preconnect): This should be no crossorigin -->
-  <link rel="preconnect" href="https://fonts.googleapis.com" crossorigin="anonymous" />
+  <link rel="preconnect" href="http://localhost:10503" crossorigin="anonymous" />
   <!-- PASS(preconnect): This uses crossorigin correctly -->
-  <link rel="preconnect" href="https://fonts.gstatic.com" crossorigin />
+  <link rel="preconnect" href="http://localhost:10420" crossorigin />
```
Comment (adamraine) [inline cli/test/fixtures/preload.html:9]: 🍃 🔥 💨

File: cli/test/fixtures/static-server.js [hunk 4]
```diff
@@ -242,25 +246,26 @@ class Server {
   }
 }
 
-const serverForOnline = new Server();
-const serverForOffline = new Server();
-
-serverForOnline._server.on('error', e => console.error(e.code, e));
-serverForOffline._server.on('error', e => console.error(e.code, e));
+async function createServers() {
+  const servers = [10200, 10503, 10420].map(port => {
```
[the remark this answers is not in the document]
Comment (connorjclark) [reply cli/test/fixtures/static-server.js:250]: I'd want something more accurate than that - it should verify that the port is actually a static server not just that something _happens_ to be on the port already.

So I'm punting for this PR, but I'll file an issue. https://github.com/GoogleChrome/lighthouse/issues/14302

File: core/test/fraggle-rock/scenarios/pptr-test-utils.js (hunk 2 above)
Comment (connorjclark) [reply core/test/fraggle-rock/scenarios/pptr-test-utils.js:42]: createServers hardcodes the ports, so can't use that here.

Comment (connorjclark) [reply core/test/fraggle-rock/scenarios/pptr-test-utils.js:42]: There's def. room for a better Server refactor here, but I don't wanna  get bogged down on it.

File: cli/test/fixtures/perf/preload_tester.js (hunk 1 above)
Comment (connorjclark) [reply cli/test/fixtures/perf/preload_tester.js:15]: I think I can drop this, but @paulirish or @adamraine please lmk if you agree

Comment (adamraine) [reply cli/test/fixtures/perf/preload_tester.js:15]: Yeah I think it should be fine without

**Review by adamraine — APPROVED** [review]

**Pull request merged**